### DISCUSSION TITLE
[Archive] AppFlowy's 6-month technical roadmap Dec 2021 - May 2022

### DISCUSSION THREAD

**Discussion by annieappflowy** [AppFlowy-IO/AppFlowy#116] [General]
![techroadmap](https://user-images.githubusercontent.com/12026239/145951724-d0add3cb-8816-42f3-a43c-9d5d4b58e4d4.jpg)

The top layer (block -> plugin market) is aligned with https://github.com/AppFlowy-IO/appflowy/projects/2
Open to feedback and participation :D

Edited: updated the roadmap based on @fuzznaut 's feedback

**Comment by nr7s** [comment]
The chart looks great, I'm very thankful that you are leading an initiative like this, it's something I've been looking forward for a while. 

Here are my 2 cents regarding the chart:
1. I'm not entirely sure what do the releases in stage 3 actually cover. I can get the general idea but I'm worried that my interpretation is not aligned with yours. Is there a document explaining the terms present in the chart with more detail? I might have missed it.
2. Given that this is a 6 month roadmap, does a start date for stage 1 already exist?
3. A simple CI setup could be done as soon as we have stable docker images or when we have some unit tests and then grow organically as the rest of the codebase grows. Were you thinking in any particular technology for the CI? Maybe this is something too specific that can be discussed further ahead.

Out of curiosity what did you use to make the chart?  Whimsical?

**EDIT:** nevermind comment 3, I went to look for a CI setup as I noticed it was mentioned on discord and found it [here](https://github.com/AppFlowy-IO/appflowy/tree/main/.github/workflows).

**Reply by annieappflowy** [reply]
> The chart looks great, I'm very thankful that you are leading an initiative like this, it's something I've been looking forward for a while.
> 
> Here are my 2 cents regarding the chart:
> 
> 1. I'm not entirely sure what do the releases in stage 3 actually cover. I can get the general idea but I'm worried that my interpretation is not aligned with yours. Is there a document explaining the terms present in the chart with more detail? I might have missed it.
> 2. Given that this is a 6 month roadmap, does a start date for stage 1 already exist?
> 3. A simple CI setup could be done as soon as we have stable docker images or when we have some unit tests and then grow organically as the rest of the codebase grows. Were you thinking in any particular technology for the CI? Maybe this is something too specific that can be discussed further ahead.
> 
> Out of curiosity what did you use to make the chart? Whimsical?
> 
> **EDIT:** nevermind comment 3, I went to look for a CI setup as I noticed it was mentioned on discord and found it [here](https://github.com/AppFlowy-IO/appflowy/tree/main/.github/workflows).

Hi fuzznaut,

Thank you for your thoughtful feedback. I'm editing the roadmap now to incorporate your suggestions.
To the question about the tool, we use https://miro.com/

Best,
Annie

**Comment by annieappflowy** [comment]
The roadmap will be updated once v0.0.4 is released 
cc: @appflowy
